### PR TITLE
ItemTypeDateRange

### PR DESCRIPTION
**Pull request by moluuser**

### CODE REVIEW

**Review by sunfmin — APPROVED** [review]

File: vuetifyx/filter_test.go [hunk 1]
```diff
@@ -84,6 +92,73 @@ var setByQueryCases = []struct {
...
+				Key:       "created",
+				ItemType:  ItemTypeDateRange,
+				Modifier:  ModifierBetween,
+				Selected:  true,
+				ValueFrom: "2019-04-10",
+				ValueTo:   "2019-04-11",
+			},
+		}),
+		expectedSQLConds: "created_at >= ? AND created_at <= ?",
+		expectedSQLArgs:  []interface{}{mustParseDatePickerValue("2019-04-10"), mustParseDatePickerValue("2019-04-11")},
```
Comment (xuxinx) [inline vuetifyx/filter_test.go:116]: `created.lte=2019-04-11&created.gte=2019-04-10` should results `created_at >= 2019-04-10 00:00 AND created_at < 2019-04-12 00:00`

File: vuetifyx/vuetifyxjs/src/components/DatePicker.vue [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+import {format, parse} from 'date-fns'
+
+const DEFAULT_DATE = ''
+const DEFAULT_DATE_FORMAT = 'yyyy-MM-dd'
+const DEFAULT_DIALOG_WIDTH = 380
+const DEFAULT_CLEAR_TEXT = 'CLEAR'
+const DEFAULT_OK_TEXT = 'OK'
+
+export default {
+	name: 'v-datepicker',
```
Comment (xuxinx) [inline vuetifyx/vuetifyxjs/src/components/DatePicker.vue:59]: vx-datepicker

File: vuetifyx/vuetifyxjs/src/components/Filter.tsx [hunk 3]
```diff
@@ -21,8 +21,10 @@ import {
 import VAutocomplete from './Autocomplete';
 import * as constants from './Constants';
 import DateTimePicker from './DateTimePicker.vue';
+import DatePicker from './DatePicker.vue';
 import { encodeFilterData, filterData } from './FilterData';
 import LinkageSelect from './LinkageSelect.vue';
+import dateTimePicker from "@/components/DateTimePicker.vue";
```
Comment (xuxinx) [inline vuetifyx/vuetifyxjs/src/components/Filter.tsx:27]: what is this?

File: vuetifyx/filter.go [hunk 4]
```diff
@@ -316,6 +316,17 @@ func (fd FilterData) SetByQueryString(qs string) (sqlCondition string, sqlArgs [
 				if it.ItemType == ItemTypeDate {
 					conds = append(conds, sqlcToCond(sqlc, "gte"), sqlcToCond(sqlc, "lt"))
 					sqlArgs = append(sqlArgs, ival, ival.(time.Time).Add(24*time.Hour))
+				} else if it.ItemType == ItemTypeDateRange {
+					if mod == "gte" {
+						conds = append(conds, sqlcToCond(sqlc, "gte"))
+						sqlArgs = append(sqlArgs, ival)
+						it.ValueFrom = val
```
Comment (xuxinx) [inline vuetifyx/filter.go:323]: > it.ValueFrom = val

move this to the below block that deals with FilterData struct things

**Review by xuxinx — APPROVED** [review]

**Pull request merged**